src/anndata/_core/merge.py

```diff
@@ -658,7 +658,12 @@ def _apply_to_array(self, el, *, axis, fill_value=None):
 
         indexer = self.idx
 
-        # Indexes real fast, and does outer indexing
+        # Fallback to numpy: keep pandas
+        # Force to NumPy (materializes JAX/Cubed); fine for small tests,
+        # but may be slow or fail on large/lazy arrays
```

Comment:
This code doesn’t just run for tests though. Also are you sure that this is a good idea for arrays with pandas dtypes?